src/nbl/asset/utils/CIESProfileParser.cpp

```diff
@@ -150,12 +157,58 @@ bool CIESProfileParser::parse(CIESProfile& result)
         }
     }
 
-    const double factor = ballastFactor * candelaMultiplier;
-    for (int i = 0; i < hSize; i++)
-        for (int j = 0; j < vSize; j++)
-            result.setValue(i, j, factor * getDouble("intensity value truncated"));
-    
-    result.maxValue = *std::max_element(std::begin(result.data), std::end(result.data));
+    {
+        const double factor = ballastFactor * candelaMultiplier;
+        for (int i = 0; i < hSize; i++)
+            for (int j = 0; j < vSize; j++)
+                result.setCandelaValue(i, j, factor * getDouble("intensity value truncated"));
+    }
+
+    float totalEmissionIntegral = 0.0, nonZeroEmissionDomainSize = 0.0;
+    constexpr auto FULL_SOLID_ANGLE = 4.0f * core::PI<float>();
+
+    if (result.symmetry == CIESProfile::ISOTROPIC && result.vAngles.size() == 1) // not sure if any IES provides this kind of distribution but it should be possible to do so meaning fixed candela value for each direction
+    {
+        const auto candelaValue = result.getCandelaValue(0, 0);
+        constexpr auto solidAngle = FULL_SOLID_ANGLE; // spans whole sphere area
+
+        totalEmissionIntegral = candelaValue * solidAngle;
+        nonZeroEmissionDomainSize = solidAngle;
+    }
+    else
+        for (size_t i = 0; i < result.hAngles.size(); i++)
+            for (size_t j = 0; j < result.vAngles.size(); j++)
+            {
+                const auto candelaValue = result.getCandelaValue(i, j);
+
+                auto getDPhiRad = [&]()
+                {
+                    if (result.symmetry == CIESProfile::ISOTROPIC) // this could be nice templated lambda, TODO: correct once c++20 enters this branch
+                        return core::PI<float>() * 2.0f; // spans whole circle
+                    else
+                        return core::radians<float>((i < result.hAngles.size() - 1) ? result.hAngles[i + 1] - result.hAngles[i] : result.hAngles[i] - result.hAngles[i - 1]);
+                };
```

Comment:
you can hoist `dPhi` computation outside the `vAngle` loop

Reply:
as a matter of readability I'd prefer
```
i ? (hAngles[i]-hAngles[i-1]):(360.f-hAngles.back()+hAngles.front())
```
to
```
(i < result.hAngles.size() - 1) ? result.hAngles[i + 1] - result.hAngles[i] : result.hAngles[i] - result.hAngles[i - 1])
```

Also I think you have a bug when `i==hangles.size()-1` you're getting the phi difference of the wrong points (last and second to last)

Reply:
aaah I forgot about domain symmetries and that `hAngles.front()=DOMAIN_START` and `hAngles.back()=DOMAIN_END`

in this case all you need is
```cpp
result.symmetry!=ISOTROPIC ? (hAngles[i+1]-hAngles[i]):twoPI;
```

and for `i` to bary from 0 to `hAngles.size()-2`

Reply:
resolved in following commit